src/pages/ContinentPage.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Link, useNavigate, useLocation, useParams,
+} from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { FaSearchLocation } from 'react-icons/fa';
+import FetchStats from '../store/api';
+import { GetStats } from '../store/reducer';
+import Pagination from '../components/Pagination';
+
+const ContinentPage = () => {
+  const CountryStore = useSelector((store) => store.details);
+  const dispatch = useDispatch();
+  const { continent } = useParams();
+
+  useEffect(() => {
+    if (CountryStore.length === 0) {
+      FetchStats()
+        .then((response) => dispatch(GetStats(response)));
+    }
+  }, []);
+
+  let Continents = CountryStore.filter((item) => item.continent === continent);
+
+  const totalPopulation = Continents.reduce((total, data) => ({
+    population: total.population + data.population,
+  }), { population: 0 });
+
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const search = query.get('search') || '';
+  Continents = Continents.filter((country) => country.country.toLowerCase().includes(search));
```

Comment:
- [ ] Again, from the requirement of this project, it will be better for us to have a stateless component that filters data retrieved from the API. What did you think about this?